controllers/configurationpolicy_controller.go

```diff
@@ -1428,6 +1469,13 @@ func enforceByCreatingOrDeleting(obj singleObject, dclient dynamic.Interface) (r
 			log.V(2).Info("Created missing must have object", "resource", obj.gvr.Resource, "name", obj.name)
 			reason = "K8s creation success"
 			msg = fmt.Sprintf("%v %v was missing, and was created successfully", obj.gvr.Resource, idStr)
+
+			var uidIsString bool
+			uid, uidIsString, err = unstructured.NestedString(obj.object.Object, "metadata", "uid")
+
+			if !uidIsString || err != nil {
+				log.Error(err, "Tried to set UID in status but the field is not a string")
```

Comment:
More accurately, it's either not a string, or some other error happened... You might consider splitting them into separate cases? But I'm pretty sure k8s will guarantee it's a string, so I don't see this really being an issue.